src/awkward/_v2/_slicing.py

```diff
@@ -59,8 +59,9 @@ def getitem_broadcast(items):
                     out[-1].metadata["shape"] = x.shape
                 elif issubclass(x.dtype.type, (np.bool_, bool)):
                     if len(x.shape) == 1:
-                        out.append(ak._v2.index.Index64(nplike.nonzero(x)[0]))
-                        out[-1].metadata["shape"] = x.shape
+                        current = ak._v2.index.Index64(nplike.nonzero(x)[0])
+                        out.append(current)
+                        out[-1].metadata["shape"] = current.data.shape
```

Comment:
That makes a lot of sense: the shape had been wrong because at some point, we made the assumption that `x` is an array of integer positions, but here it's an array of booleans, unlike what goes into `out`. If this was the case, the following should have worked (pre-bug-fix), because the shape of the booleans and the shape of the integers is the same:

```python
>>> ak._v2.Array([[1.1, 2.2, 3.3], [4.4], [5.5, 6.6]])[[True, True, True], 0]
<Array [1.1, 4.4, 5.5] type='3 * float64'>
```

and it did.